build.gradle.kts

```
@@ -0,0 +1,303 @@
+    version.set(intellijVersions["buildTarget"] as String)
+    plugins.set(listOf("git4idea")) // Needed solely for ArchUnit
+}
+
+if (!isCI) {
+    // The output of this task is for some reason very poorly cached,
+    // and the task takes a significant amount of time,
+    // while the index of searchable options is of little use for local development.
+    tasks.withType<BuildSearchableOptionsTask> {
+        enabled = false
+    }
+
+}
+
+tasks.withType<PatchPluginXmlTask> {
+    // `sinceBuild` is exclusive when we are using `*` in version but inclusive when without `*`
+    sinceBuild.set(IntellijVersionHelper.toBuildNumber(intellijVersions["earliestSupportedMajor"] as String))
+
+    // In `untilBuild` situation is inverted: it's inclusive when using `*` but exclusive when without `*`
+    untilBuild.set(IntellijVersionHelper.toBuildNumber(intellijVersions["latestSupportedMajor"] as String) + ".*")
+
+    // Note that the first line of the description should be self-contained since it is placed into embeddable card:
+    // see e.g. https://plugins.jetbrains.com/search?search=git%20machete
+    pluginDescription.set(file("$rootDir/DESCRIPTION.html").readText())
+
+    changeNotes.set("<h3>v${rootProject.version}</h3>\n\n" + file("$rootDir/CHANGE-NOTES.html").readText())
+}
+
+tasks.withType<RunIdeTask> {
+    maxHeapSize = "4G"
+}
+
+tasks.withType<RunPluginVerifierTask> {
+    val maybeEap = if (intellijVersions["eapOfLatestSupportedMajor"] != null) listOf((intellijVersions["eapOfLatestSupportedMajor"] as String).replace("-EAP-(CANDIDATE-)?SNAPSHOT", ""))
+    else emptyList()
+    ideVersions.set(listOf(*(intellijVersions["latestMinorsOfOldSupportedMajors"] as List<String>).toTypedArray(), intellijVersions["latestStable"] as String, *maybeEap.toTypedArray()))
+
+    val skippedFailureLevels = `java.util`.EnumSet.of(DEPRECATED_API_USAGES, EXPERIMENTAL_API_USAGES, NOT_DYNAMIC, SCHEDULED_FOR_REMOVAL_API_USAGES)
+    failureLevel.set(`java.util`.EnumSet.complementOf(skippedFailureLevels))
+}
+
+tasks.withType<PublishPluginTask> {
+    token.set(jetbrainsMarketplaceToken)
+}
+
+val uiTest = sourceSets.create("uiTest")
+
+val uiTestImplementation: Configuration by configurations.getting {
+    extendsFrom(configurations.testImplementation.get())
+}
+
+val uiTestRuntimeOnly: Configuration by configurations.getting {
+    extendsFrom(configurations.testRuntimeOnly.get())
+}
+
+val uiTestTargets = if (project.properties["against"] != null) IntellijVersionHelper.resolveIntelliJVersions(project.properties["against"] as String) else listOf(intellijVersions["buildTarget"])
+
+uiTestTargets.onEach { version ->
+    tasks.register<Test>("uiTest_${version}") {
+        description = "Runs UI tests."
+        group = "verification"
+
+        testClassesDirs = uiTest.output.classesDirs
+        classpath = configurations["uiTestRuntimeClasspath"] + uiTest.output
+
+        val bp = dependsOn(":buildPlugin")
```

```suggestion
        val buildPlugin = tasks.findByPath(":buildPlugin")!!
```

This alone should fix your problem with `newFileSystem` in UI tests. Currently, `ui-test.plugin.path` points to some random **directory**, not plugin zip file.

---

Opened https://github.com/VirtusLab/ide-probe/issues/234 btw